File: java/src/main/java/com/files/models/UsageDailySnapshot.java

package com.files.models;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.HashMap;

public class UsageDailySnapshot
{
    private HashMap<String, Object> attributes;
    private HashMap<String, Object> options;

    public UsageDailySnapshot(HashMap<String, Object> attributes, HashMap<String, Object> options)
    {
        this.attributes = attributes;
        this.options = options;
    }

    /**
     * ID of the usage record
     */
    @Getter
    public Long id;

    /**
     * The date of this usage record
     */
    @Getter
    public LocalDate date;

    /**
     * The quantity of storage held for this site
     */
    @Getter
    public Long currentStorage;

    /**
     * Usage broken down by each top-level folder
     */
    @Getter
    public Object[] usageByTopLevelDir;



    /**
     * Parameters:
     *   page - integer - Current page number.
     *   per_page - integer - Number of records to show per page.  (Max: 10,000, 1,000 or less is recommended).
     *   action - string - Deprecated: If set to `count` returns a count of matching records rather than the records themselves.
     */
    // TODO: Use types for path_and_primary_params
    public static UsageDailySnapshot list( HashMap<String, Object> parameters, HashMap<String, Object> options)
    {
        parameters = parameters != null ? parameters : new HashMap<String, Object>();
        options = options != null ? options : new HashMap<String, Object>();

        if (parameters.containsKey("page") && !(parameters.get("page") instanceof Long ))
        {
            throw new IllegalArgumentException("Bad parameter: page must be of type Long parameters[\"page\"]");
        }
        if (parameters.containsKey("per_page") && !(parameters.get("per_page") instanceof Long ))
        {
            throw new IllegalArgumentException("Bad parameter: per_page must be of type Long parameters[\"per_page\"]");
        }
        if (parameters.containsKey("action") && !(parameters.get("action") instanceof String ))
        {
            throw new IllegalArgumentException("Bad parameter: action must be of type String parameters[\"action\"]");
        }
        // TODO: Send request
        return null;
    }

    public static UsageDailySnapshot all(HashMap<String, Object> parameters, HashMap<String, Object> options)
    {
        return list(parameters, options);
    }
}

File: dotnet/sdk/Files.com/Models/UsageDailySnapshot.cs

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Files.com.Models
{
    public class UsageDailySnapshot
    {
        private Dictionary<string, object> attributes;
        private Dictionary<string, object> options;

        public UsageDailySnapshot()
        {
            this.attributes = new Dictionary<string, object>();
            this.options = new Dictionary<string, object>();

            this.attributes.Add("id", null);
            this.attributes.Add("date", null);
            this.attributes.Add("current_storage", null);
            this.attributes.Add("usage_by_top_level_dir", new object[0]);
        }

        public UsageDailySnapshot(Dictionary<string, object> attributes, Dictionary<string, object> options)
        {
            this.attributes = attributes;
            this.options = options;
        }

        /// <summary>
        /// ID of the usage record
        /// </summary>
        [JsonPropertyName("id")]
        public Nullable<Int64> Id
        {
            get { return (Nullable<Int64>) attributes["id"]; }
        }

        /// <summary>
        /// The date of this usage record
        /// </summary>
        [JsonPropertyName("date")]
        public Nullable<DateTime> Date
        {
            get { return (Nullable<DateTime>) attributes["date"]; }
        }

        /// <summary>
        /// The quantity of storage held for this site
        /// </summary>
        [JsonPropertyName("current_storage")]
        public Nullable<Int64> CurrentStorage
        {
            get { return (Nullable<Int64>) attributes["current_storage"]; }
        }

        /// <summary>
        /// Usage broken down by each top-level folder
        /// </summary>
        [JsonPropertyName("usage_by_top_level_dir")]
        public object[] UsageByTopLevelDir
        {
            get { return (object[]) attributes["usage_by_top_level_dir"]; }
        }



        /// <summary>
        /// Parameters:
        ///   page - integer - Current page number.
        ///   per_page - integer - Number of records to show per page.  (Max: 10,000, 1,000 or less is recommended).
        ///   action - string - Deprecated: If set to `count` returns a count of matching records rather than the records themselves.
        /// </summary>
        /// TODO: Use types for path_and_primary_params
        /// TODO: Get return values from the operation (either model.class_name or an array of them)
        public static async Task<Object> List(
            
            Dictionary<string, object> parameters = null,
            Dictionary<string, object> options = null
        )
        {
            parameters = parameters != null ? parameters : new Dictionary<string, object>();
            options = options != null ? options : new Dictionary<string, object>();

            if (parameters.ContainsKey("page") && !(parameters["page"] is Nullable<Int64> ))
            {
                throw new ArgumentException("Bad parameter: page must be of type Nullable<Int64>", "parameters[\"page\"]");
            }
            if (parameters.ContainsKey("per_page") && !(parameters["per_page"] is Nullable<Int64> ))
            {
                throw new ArgumentException("Bad parameter: per_page must be of type Nullable<Int64>", "parameters[\"per_page\"]");
            }
            if (parameters.ContainsKey("action") && !(parameters["action"] is string ))
            {
                throw new ArgumentException("Bad parameter: action must be of type string", "parameters[\"action\"]");
            }

            string responseJson = await FilesApi.SendRequest("/usage_daily_snapshots", HttpMethod.Get, parameters, options);

            return JsonSerializer.Deserialize<UsageDailySnapshot[]>(responseJson);
        }

        /// TODO: Get return values from the operation (either model.class_name or an array of them)
        public static async Task<Object> All(
            
            Dictionary<string, object> parameters = null,
            Dictionary<string, object> options = null
        )
        {
            return await List(parameters, options);
        }

    }
}

